tests/test_entities.py

```diff
@@ -2204,6 +2230,69 @@ def test_creation_and_update(self):
         self.assertEqual(expected_dct, cfg_template.update_payload())
 
 
+class TemplateInputTestCase(TestCase):
+    """Tests for :class:`nailgun.entities.TemplateInput`."""
+    def setUp(self):
+        self.entity = entities.TemplateInput(config.ServerConfig('some url'),
+                                             template='1')
+        self.read_patcher = mock.patch.object(entities.TemplateInput, 'read')
+        self.read = self.read_patcher.start()
+        self.read.return_value = entities.TemplateInput(
+            self.entity,
+            template=1,
+            name='my new template input',
+            input_type='user')
+        self.read_json_patcher = mock.patch.object(
+            EntityReadMixin,
+            'read_json')
+        self.read_json = self.read_json_patcher.start()
+
+    def tearDown(self):
+        self.read_json_patcher.stop()
+        self.read_patcher.stop()
+
+    def test_read(self):
+        entities.TemplateInput(
+            config.ServerConfig('some url'),
+            id='1',
+            name='my new template input',
+            template='1'
+        ).read(attrs={
+            'name': 'my new template input',
+            'template': '1',
+            'input_type': 'user'
+        })
```

Comment:
What is the purpose of this test?

Reply:
perhaps we have to check that read json was invoked and with some expected prams.

Reply:
yes, maybe we should mock only read_json

Reply:
suggest the following testcase for this entity read
```python
class TemplateInputTestCase(TestCase):
    """Tests for :class:`nailgun.entities.TemplateInput`."""
    def setUp(self):
        self.cfg = config.ServerConfig('some url')
        self.job_template = entities.JobTemplate(self.cfg, id=2)
        self.entity = entities.TemplateInput(self.cfg, id=1, template=self.job_template)
        self.data = dict(
            id=1,
            description=None,
            fact_name=None,
            input_type='user',
            name='my new template input',
            options=None,
            puppet_class_name=None,
            puppet_parameter_name=None,
            required=False,
            template_id=self.job_template.id,
            variable_name=None,
        )
        self.read_json_patcher = mock.patch.object(EntityReadMixin, 'read_json')
        self.read_json = self.read_json_patcher.start()
        self.read_json.return_value = self.data.copy()
        del self.data['template_id']

    def tearDown(self):
        self.read_json_patcher.stop()

    def test_read(self):
        entity = self.entity.read()
        self.read_json.assert_called_once()
        self.assertEqual(
            self.data,
            {key: getattr(entity, key) for key in self.data.keys()}
        )
        self.assertIsInstance(entity.template, entities.JobTemplate)
        self.assertEqual(entity.template.id, self.job_template.id)
```

Reply:
nice, thank you :)